solver/src/main/java/ConstraintSolver.java

```diff
@@ -0,0 +1,47 @@
+import org.testcontainers.containers.Container;
+import org.testcontainers.containers.GenericContainer;
+import org.testcontainers.images.builder.ImageFromDockerfile;
+import org.testcontainers.containers.BindMode;
+
+import java.io.IOException;
+
+public class ConstraintSolver implements AutoCloseable  {
+
+    public static final String Z3_DOCKER_IMAGE = "ghcr.io/z3prover/z3:ubuntu-20.04-bare-z3-sha-ba8d8f0";
+
+    // The default entrypoint runs directly z3, we need to keep it running
+    public static final String ENTRYPOINT = "while :; do sleep 1000 ; done";
+
+    private final String containerPath = "/smt2-resources/";
+    private final GenericContainer<?>  z3Prover;
+
+    public ConstraintSolver() {
+
+        String resourcesFolder = System.getProperty("user.dir") + "/src/test/resources/";
+
+        ImageFromDockerfile image = new ImageFromDockerfile()
+                .withDockerfileFromBuilder(builder -> builder
+                        .from(Z3_DOCKER_IMAGE)
+                        .entryPoint(ENTRYPOINT)
+                        .build());
+
+        z3Prover = new GenericContainer<>(image)
+                        .withFileSystemBind(resourcesFolder, containerPath, BindMode.READ_WRITE);
```

Comment:
why binding to actual FS? does solver write to disk, and we must read from disk to get the solver's results? if so, clarify

Reply:
The current implementation of the Z3 solver I'm using reads content either from STDIN or from a file. Since I couldn't find a way to send data through STDIN every time I want to execute a command, I've opted for the second option. Additionally, I need to dynamically generate various inputs, which is why I've linked the file system folder to the Docker volume. I then confirmed the functionality by testing the creation of a new file, and it worked as expected.

In summary, I'm reading input from a file, and the result is returned in STDOUT. Does that make sense? If you agree, I can also document this in the code for added clarity.

Reply:
hi. thanks. yes, but add this kind of info here in the code as a comment. 

for this prototype, we do not care too much about performance. but, in the long run, an approach that requires to communicate with an external process via files is not efficient.

Reply:
maybe use `ConstraintSolver` as an abstract class or interface, and actual implementation (eg `ConstraintSolverZ3InDocker`) as a subclass

Reply:
Agree! I've added a comment explaining this and also created the ConstraintSolver interface. Anyway, for now, the solve method receives the input file name as a parameter, the idea is to refactor that to receive a set of constraints and will do that on another PR.